### ISSUE TITLE
[Steam] Renpy games not working?

### ISSUE THREAD

**Issue by ghost** [flathub/com.valvesoftware.Steam#54]
Hello Everyone!
I have been trying to run a few Renpy based games in Steam under Flatpak, but I keep getting an error,
`ImportError: /home/gego/.var/app/com.valvesoftware.Steam/data/Steam/steamapps/common/[game]/lib/linux-x86_64/lib/python2.7/_ctypes.so: undefined symbol: PyUnicodeUCS4_FromEncodedObject`, and I don't know what is wrong.

It is strange, this seems to be a problem with a many games: The Sakura Series[1], The Sunrider Series[2], The (only) World End Economica (that has a Linux release), **but not** Long Live The Queen (an older game).

But I can start the games from the command line, by cd:ing into the directory and running it manually.

Anyone else have these problems?

[1] Tested: Beach 1 & 2, Space, Spirit
[2] The first part is Free.
```
$ flatpak --version 
Flatpak 0.10.2
$ uname -a
Linux octo 4.10.0-42-generic #46-Ubuntu SMP Mon Dec 4 14:38:01 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release --all
LSB Version:	core-2.0-amd64:core-2.0-noarch:core-3.0-amd64:core-3.0-noarch:core-3.1-amd64:core-3.1-noarch:core-3.2-amd64:core-3.2-noarch:core-4.0-amd64:core-4.0-noarch:core-4.1-amd64:core-4.1-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 17.04
Release:	17.04
Codename:	zesty
```

**Comment by TingPing** [comment]
So googling around it seems like this error happens with two Python versions are mixed up and one is built with `--enable-unicode=ucs4` and the other not. Given the path I would call it a bug in the game that they didn't correctly bundle Python.

**Comment by GegoXaren** [comment]
But that does not explain why I can start it from the terminal using the script. 
I will check tomorrow if it runs in a non-flatpak enviorment.

And I highley dought that every Renpy game exibits the exact same problem due to packning issues and noone would notice? Perhaps I should ask on the games Steam forum? 

Also, I think it worked a few months back, (in flatpak) before I upgraded to 17.04...

**Comment by TingPing** [comment]
> But I can start the games from the command line, by cd:ing into the directory and running it manually.

As in.. `flatpak run --command=sh ...` ?

> And I highley dought that every Renpy game exibits the exact same problem due to packning issues and noone would notice? 

I'd guess Renpy has common tooling for packaging. A huge percentage of games shipped on Linux are flat out broken because nobody tests or cares about anything other than a single distro.

> Also, I think it worked a few months back, (in flatpak) before I upgraded to 17.04...

Probably a change in this package rather than distro.

**Comment by GegoXaren** [comment]
> As in.. flatpak run --command=sh com. ?

Yes, it starts:
```
$ cd data/Steam/steamapps/common/Sakura\ Sp
Sakura Space/  Sakura Spirit/ 
sh-4.3$ cd data/Steam/steamapps/common/Sakura\ Spirit/
sh-4.3$ ls
game  log.txt	   renpy	      Sakura Spirit.exe  Sakura Spirit.sh
lib   README.html  Sakura Spirit.app  Sakura Spirit.py
sh-4.3$ ./
README.html        Sakura Spirit.exe  Sakura Spirit.sh   lib/
Sakura Spirit.app/ Sakura Spirit.py   game/              renpy/
sh-4.3$ ./Sakura\ Spirit.sh
```
![skarmbild fran 2017-12-20 12-46-25](https://user-images.githubusercontent.com/8343266/34205722-cf959274-e583-11e7-9540-996c10fc47d9.png)

**Comment by GegoXaren** [comment]
And just to confirm:
It does run in the non-flatpak version of Steam.

**Comment by TingPing** [comment]
> sh-4.3$ ./Sakura\ Spirit.sh

Well that would bypass the Steam runtime, but I'm not sure why it would work on the host if the Steam runtime is the problem.

**Comment by nanonyme** [comment]
Just checking: does this still fail?

**Comment by nanonyme** [comment]
This should now work, the Python shipped to fix #152 waw compiled with UCS4 support. I'm going to close this as fixed, please re-open if problem persists
